sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -97,6 +102,12 @@
             .typeSystem(connection.getTypeFactory().getTypeSystem())
             .operatorTable(ChainedSqlOperatorTable.of(opTab0, catalogReader))
             .build();
+
+    options = PipelineOptionsFactory.create();
+  }
+
+  public void setOptions(PipelineOptions options) {
```

Comment:
I would argue this should be more like `setPlannerFactory()` taking either a `String` or `Constructor`. I like the `PipelineOptions` approach, I just don't think they belong in `BeamQueryPlanner`.

Reply:
I updated this PR and keep/pass in `Constructor` to `BeamQueryPlanner`.